polynote-frontend/polynote/ui/input/plot_selector.ts

```diff
@@ -0,0 +1,1238 @@
+                axis: {
+                    title: plot.x.title
+                }
+            },
+            // TODO: disabling color series because it doesn't work well.
+            // color: mapOpt(
+            //     plot.color,
+            //     colorAxis => ({
+            //         field: colorAxis.field,
+            //         type: dimensionType(schema.fieldType(colorAxis.field)!),
+            //         title: colorAxis.title
+            //     })
+            // )
+        },
+        layer: [
+            {
+                mark: { type: "rule", style: "boxplot-rule" },
+                encoding: {
+                    y: { field: `${baseY}.min`, type: "quantitative", title: plot.y.title || plot.y.field, scale: { zero: plotDef.forceZero }},
+                    y2: { field: `${baseY}.max` }
+                }
+            },
+            {
+                mark: { type: "bar", style: "boxplot-box", size, tooltip: { content: 'data' } },
+                encoding: {
+                    y: { field: `${baseY}.q1`, type: "quantitative" },
+                    y2: { field: `${baseY}.q3` },
+                }
+            },
+            {
+                mark: { type: "tick", color: "white", style: 'boxplot-median', size },
+                encoding: {
+                    y: { field: `${baseY}.median`, type: "quantitative"}
+                }
+            },
+            {
+                mark: { type: "point", color: "black" },
+                encoding: {
+                    y: { field: `${baseY}.mean`, type: "quantitative" }
+                }
+            }
+        ]
+    };
+
+    // TODO: encode color doesn't really work. Make it work.
+
+    return base;
+}
+
+function piePlot(plotDef: PlotDefinition, plot: PiePlot, schema: StructType): TopLevelSpec {
+    const schemaField = schema.fields.find(field => field.name === plot.x.field);
+    if (!schemaField) {
+        throw new Error(`Field ${plot.x.field} is not in the schema`);
+    }
+    return {
+        $schema: 'https://vega.github.io/schema/vega-lite/v4.json',
+        title: plotDef.title,
+        data: {name: plotDef.value},
+        mark: {
+            type: 'arc',
+            tooltip: { content: 'data' }
+        },
+        encoding: {
+            color: {
+                field: plot.x.field,
+                type: 'nominal',
+                legend: {
+                    title: plot.x.title
+                }
+            },
+            theta: {
+                field: `sum(${plot.y.field})`,
+                type: 'quantitative'
+            }
+        }
+    };
+}
+
+function histogramPlot(plotDef: PlotDefinition, plot: HistogramPlot, schema: StructType): TopLevelSpec {
+    const schemaField = schema.fields.find(field => field.name === plot.x.field);
+    if (!schemaField) {
+        throw new Error(`Field ${plot.x.field} is not in the schema`);
+    }
+
+    return {
+        $schema: 'https://vega.github.io/schema/vega-lite/v4.json',
```

Comment:
Any impact of this schema being v4 rather than v3?

Reply:
it's newer?